pkg/chunk/cached_store.go

```diff
@@ -945,44 +945,44 @@ func parseObjOrigSize(key string) int {
 	return l
 }
 
-func (store *cachedStore) uploadStagingFile(key string, stagingPath string) {
+func (store *cachedStore) uploadStagingFile(key string, stagingPath string) error {
 	store.currentUpload <- true
 	defer func() {
 		<-store.currentUpload
 	}()
 
 	if !store.canUpload() {
-		return
+		return nil
 	}
 	store.pendingMutex.Lock()
 	item, ok := store.pendingKeys[key]
 	store.pendingMutex.Unlock()
 	if !ok {
```

Comment:
maybe `defer item.uploading = false` is enough, no matter success or failed, we just want to filter the duplicated process.